crates/test-cluster/src/lib.rs

```diff
@@ -1087,20 +1107,50 @@ impl TestClusterBuilder {
             }));
         }
 
+        let mut temp_data_ingestion_dir = None;
+        let mut data_ingestion_path = None;
+
+        if self.indexer_backed_rpc {
+            if self.data_ingestion_dir.is_none() {
```

Comment:
Could this code be moved to line 1138?

Reply:
It cannot. `self.start_swarm()` right after this block uses `self.data_ingestion_dir`.